static/src/javascripts/projects/common/modules/experiments/low-friction-participation.js

```diff
@@ -0,0 +1,261 @@
+                templateVars.buttonText = 'Choose ' + buttonTextVal;
+                templateVars.elType = 'button';
+            } else {
+                templateVars.elType = 'span';
+            }
+
+            buttonString += template(lowFrictionButtons, merge(settings.templateVars, templateVars));
+        }
+
+        return buttonString;
+    };
+
+    // Rendering
+
+    var render = function (state) {
+
+        var view;
+
+        if (state.complete) {
+
+            view = template(lowFrictionComplete, merge(settings.templateVars, {
+                buttons: createButtons(state)
+            }));
+
+            // Remove bindings
+            tearDown();
+
+        }  else if (state.confirming) {
+
+            view = template(lowFrictionConfirming, merge(settings.templateVars, {
+                buttons: createButtons(state)
+            }));
+
+        } else {
+
+            view = template(lowFrictionInitial, merge(settings.templateVars, {
+                buttons: createButtons(state)
+            }));
+
+        }
+
+
+        if (state.initialRender) {
+            var fullView = template(lowFrictionWrapper, merge(settings.templateVars, {
+                contents: view
+            }));
+
+            fastdomPromise.write(function() {
+                els.$articleBody.append(fullView);
+                els.$lowFricContents = $('.js-participation-low-friction__contents');
+            });
+
+        } else {
+            fastdomPromise.write(function() {
+                els.$lowFricContents.html(view);
+
+                if (state.confirming) {
+                    // Move focus to the confirm button
+                    $('.js-participation-low-fric__confirm').focus();
+                }
+            });
+        }
+
+    };
+
+    // State Handling
+
+    var updateState = function (state) {
+        // Render with merged state
+        render(merge(currentState, state));
+    };
+
+    // Getters
+
+    var getUserVote = function () {
+        if (!storage.local.isStorageAvailable()) {
+            return 'no-storage';
+        }
+
+        var currentPage = config.page.pageId,
+            votedPages = JSON.parse(storage.local.get(prefs));
+
+        // Will return result for current page if available
+        return votedPages && votedPages[currentPage];
+
+    };
+
+    // Setters
+
+    var setUserVote = function () {
+        var currentPage = config.page.pageId,
+            votedPages = JSON.parse(storage.local.get(prefs));
+
+        // If the prefs object doesn't exist, lets create one
+        if (!votedPages) {
+            votedPages = {};
+        }
+
+        votedPages[currentPage] = currentState.selectedItem;
+
+        storage.local.set(prefs, JSON.stringify(votedPages));
+    };
+
+    // Binding & Events
+
+    var itemClicked = function (event) {
+        updateState({
+            confirming: true,
+            selectedItem: $(event.currentTarget).data().itemId,
+            initialRender: false
```

Comment:
It's not a huge issue for experiment code, but I'm not sure I'd make these functions poke around internal state vars like `initialRender`. But - for an A/B test it hardly matters.